compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -397,17 +398,37 @@ public void visit(BLangRecordLiteral recordLiteral) {
 
         List<BType> matchedTypeList = getRecordCompatibleType(expType, recordLiteral);
 
+        boolean hasInvalidFieldIndex = false;
         if (matchedTypeList.isEmpty()) {
             dlog.error(recordLiteral.pos, DiagnosticCode.INVALID_LITERAL_FOR_TYPE, expType);
         } else if (matchedTypeList.size() > 1) {
             dlog.error(recordLiteral.pos, DiagnosticCode.AMBIGUOUS_TYPES, expType);
         } else {
-            recordLiteral.keyValuePairs
-                    .forEach(keyValuePair -> checkRecLiteralKeyValue(keyValuePair, matchedTypeList.get(0)));
+            for (BLangRecordKeyValue keyValuePair : recordLiteral.keyValuePairs) {
+                BType type = checkRecLiteralKeyValue(keyValuePair, matchedTypeList.get(0));
+                hasInvalidFieldIndex = type.tag == TypeTags.ERROR;
+            }
             actualType = matchedTypeList.get(0);
         }
 
         resultType = types.checkType(recordLiteral, actualType, expType);
+
+        if (recordLiteral.type.tag == TypeTags.RECORD && !hasInvalidFieldIndex) {
```

Comment:
Should we include that? `hasInvalidFieldIndex` true means there was a type checking error in the fields.

Reply:
I think we can include those as well. `hasInvalidFieldIndex` will indicate there's a type-error/non-exixsting field, in the literal. But IMO that will not harm checking for missing fields, because here we only check for the existence  of the field names.